src/Domain/CoordinateMaps/TimeDependent/Shape.cpp

```diff
@@ -0,0 +1,361 @@
+                                      extended_coefs, interpolation_info);
+  // Calculates the Pfaffian derivative at the internal collocation points of
+  // YlmSpherePack. We can't interpolate these directly as they are not smooth
+  // across the poles, so we convert them to the Cartesian gradients first,
+  // which are smooth.
+  const auto angular_gradient =
+      extended_ylm.gradient_from_coefs(extended_coefs);
+
+  tnsr::i<DataVector, 3, Frame::Inertial> cartesian_gradient(
+      extended_ylm.physical_size());
+
+  // Re-use allocations
+  std::array<DataVector, 2> collocation_theta_phis{};
+  collocation_theta_phis[0].set_data_ref(&get<2>(cartesian_gradient));
+  collocation_theta_phis[1].set_data_ref(&get<1>(cartesian_gradient));
+  collocation_theta_phis = extended_ylm.theta_phi_points();
+
+  const auto& col_thetas = collocation_theta_phis[0];
+  const auto& col_phis = collocation_theta_phis[1];
+
+  // The Cartesian derivative is the Pfaffian derivative multiplied by the
+  // inverse Jacobian matrix. Some optimizations here may be possible by
+  // introducing temporaries for some of the sin/cos which are computed twice,
+  // if the compiler CSE doesn't take care of it.
+  get<0>(cartesian_gradient) =
+      (cos(col_thetas) * cos(col_phis) * get<0>(angular_gradient) -
+       sin(col_phis) * get<1>(angular_gradient));
+
+  get<1>(cartesian_gradient) =
+      (cos(col_thetas) * sin(col_phis) * get<0>(angular_gradient) +
+       cos(col_phis) * get<1>(angular_gradient));
+
+  get<2>(cartesian_gradient) = -sin(col_thetas) * get<0>(angular_gradient);
+
+  tnsr::Ij<tt::remove_cvref_wrap_t<T>, 3, Frame::NoFrame> result(
+      get_size(centered_coords[0]));
+
+  // re-use allocation
+  auto& target_gradient_x = get<2, 0>(result);
+  auto& target_gradient_y = get<2, 1>(result);
+  auto& target_gradient_z = get<2, 2>(result);
+
+  // interpolate the cartesian gradient to the thetas and phis of the
+  // `source_coords`
+  extended_ylm.interpolate(make_not_null(&target_gradient_x),
+                           get<0>(cartesian_gradient).data(),
+                           interpolation_info);
+  extended_ylm.interpolate(make_not_null(&target_gradient_y),
+                           get<1>(cartesian_gradient).data(),
+                           interpolation_info);
+  extended_ylm.interpolate(make_not_null(&target_gradient_z),
+                           get<2>(cartesian_gradient).data(),
+                           interpolation_info);
+
+  // re-use allocation
+  auto& transition_func = get<1>(theta_phis);
+  transition_func = transition_func_->operator()(centered_coords);
+  const auto transition_func_gradient =
+      transition_func_->gradient(centered_coords);
+  // leave it to the domain specific transition function to divide by the radii
+  // as it can safely do this
+  const auto transition_func_over_radius =
+      transition_func_->map_over_radius(centered_coords);
+
+  get<0, 0>(result) = -gsl::at(centered_coords, 0) *
```

Comment:
(optional) you can use `centered_coords[0]` here without clang-tidy complaining I think, or even better `const auto& [x_centered, y_centered, z_centered] = centered_coords`.